CONTRIBUTING.md

```diff
@@ -0,0 +1,143 @@
+do genuinely want to thank you for contributing.
+
+This document contains some guidance and steps for contributing. Make sure you
+read it thoroughly, because if you miss some of these steps we will have to ask
+you to do them before we can merge it.
+
+## Before contributing: The Contributor License Agreement
+
+If you plan to contribute in the form of documentation or code, we need you to
+sign our Contributor License Agreement before we can accept your contribution.
+This covers the legalities, confirming that you retain ownership of your
+contributions and that you are licensing them to us.
+
+For some fairly complicated legal reasons we actually have three CLAs.  Which
+of the CLAs you sign depends upon whose behalf you are contributing:
+
+- All individuals (except employees of the U.S. government – see below) should
+  sign the [Individual Contributor License Agreement](http://www.projectcalico.org/community/Individual-Contributor-Agreement).
+- If you are contributing on behalf of a company or organization, *you still
+  need to sign the Individual Contributor License Agreement* as above, but
+  someone at your company or organization also needs to sign the
+  [Corporate Contributor License Agreement](http://www.projectcalico.org/community/corporate-contributor-agreement/)
+  providing a list of people authorized to commit code to Project Calico.
+- Employees of the U.S. government do not sign the Individual Contributor
+  License Agreement.  Instead, someone with authority to sign on behalf of your
+  agency should sign the [U.S Government Contributor License Agreement](http://www.projectcalico.org/community/us-government-contributor-agreement).
+
+It is important to reiterate: until you sign the required agreements we cannot
+accept your contribution!
+
+## Mailing lists and chat
+
+A great way to talk to us, ask questions, discuss features and bounce ideas
+around is to join one of the channels listed below:
+
+* [Technical Mailing List](http://lists.projectcalico.org/mailman/listinfo/calico-tech_lists.projectcalico.org)
+* [Slack Calico Users Channel](https://calicousers.slack.com) ([Sign up](https://calicousers-slackin.herokuapp.com))
+* IRC - [#calico](https://kiwiirc.com/client/irc.freenode.net/#calico)
+
+## Reporting issues
+
+Before raising an issue with the *Calico Libnetwork plugin*), please check for 
+duplicate issues, and read our [Troubleshooting](https://github.com/projectcalico/calico-containers/blob/master/docs/Troubleshooting.md)
+and our [Frequently Asked Questions](https://github.com/projectcalico/calico-containers/blob/master/docs/FAQ.md)
+documents.
+
+If you have a question, please hop on to our IRC or Slack channels (see above).
+
+If you do need to raise an issue, please include any of the following that may
+be relevant to your problem (including too much information is never
+a bad thing):
+
+-  A detailed description of the problem.
```

Comment:
docker version is important here - and possibly get them to ps the docker daemon to check it was started with a cluster store.